### PR TITLE
Fix broken links in docs and issues of links to English docs appearing in Chinese docs

### PR DESCRIPTION
**Pull request by ultmaster**
#1403

### CODE REVIEW

File: docs/zh_CN/Release.md [hunk 1]
```diff
@@ -93,8 +93,8 @@
 ### 文档
 
 * 发布中文文档网站：https://nni.readthedocs.io/zh/latest/
```
[the remark this answers is not in the document]
Comment (ultmaster) [reply docs/zh_CN/Release.md]: fixed

File: tools/nni_annotation/README.md [hunk 2]
```diff
@@ -83,10 +83,10 @@ h_pooling = max_pool(hidden_layer, pool_size)
 
    `'''@nni.report_intermediate_result(metrics)'''`
 
-`@nni.report_intermediate_result` is used to report intermediate result, whose usage is the same as `nni.report_intermediate_result` in [Trials.md](https://nni.readthedocs.io/en/latest/Trials.html)
+`@nni.report_intermediate_result` is used to report intermediate result, whose usage is the same as `nni.report_intermediate_result` in [Trials.md](https://nni.readthedocs.io/en/latest/TrialExample/Trials.html)
```
Comment (QuanluZhang) [inline tools/nni_annotation/README.md]: in [Trials.md] -> in the doc of [Write a trial run on NNI]

File: tools/nni_annotation/README.md [hunk 3]
```diff
@@ -83,10 +83,10 @@ h_pooling = max_pool(hidden_layer, pool_size)
...
 
 ### 4. Annotate final result
 
    `'''@nni.report_final_result(metrics)'''`
 
-`@nni.report_final_result` is used to report the final result of the current trial, whose usage is the same as `nni.report_final_result` in [Trials.md](https://nni.readthedocs.io/en/latest/Trials.html)
+`@nni.report_final_result` is used to report the final result of the current trial, whose usage is the same as `nni.report_final_result` in [Trials.md](https://nni.readthedocs.io/en/latest/TrialExample/Trials.html)
```
Comment (QuanluZhang) [inline tools/nni_annotation/README.md]: in [Trials.md] -> in the doc of [Write a trial run on NNI]

Comment (ultmaster) [reply tools/nni_annotation/README.md]: fixed

**Pull request merged**